src/Storages/Hive/StorageHive.cpp

```diff
@@ -381,6 +389,14 @@ class StorageHiveSource : public ISource, WithContext
                 continue;
             }
 
+            // Enrich virtual column _tags
+            if (column.name == "_tags")
```

Comment:
Hm, why does it only touches the Hive?

I think you need to modify VirtualColumnsUtils for these, same as here https://github.com/ClickHouse/ClickHouse/pull/65386